src/Workspaces/Core/Portable/Indentation/IInferredIndentationService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Indentation
+{
+    internal interface IInferredIndentationService : IWorkspaceService
+    {
+        Task<DocumentOptionSet> GetDocumentOptionsWithInferredIndentationAsync(Document document, bool explicitFormat, CancellationToken cancellationToken);
```

Comment:
📝 In a follow-up, we could probably separate the concepts of getting documentation options and getting inferred indentation. However, I don't want to see that refactoring as part of _this_ pull request.